cpp/data_structures/dynamic_tree/dtree_test.cpp

```diff
@@ -0,0 +1,50 @@
+#include "./dynamic_tree.hpp"
+
+    // just a simple tests for dynamic tree "functionality"
+int main() {
+
+    // create dynamic tree
+    cpstl::DynamicTree<int> dtree("root",1);
+
+    // insert new nodes & check
+    if(!dtree.current_node->insert("a3",1)) std::cout << "insert error !!! \n";
+    if(!dtree.current_node->insert("a2",2)) std::cout << "insert error !!! \n";
+    
+    // move and insert
+    dtree.go_to("a3");
+    if(!dtree.current_node->insert("g2",3)) std::cout << "insert error !!! \n";
+    if(!dtree.current_node->insert("g4",4)) std::cout << "insert error !!! \n";
+    
+    // move and insert
+    dtree.go_to("g2");
```

Comment:
There is a test framework implemented in the `cpstl` and this is an example https://github.com/vincenzopalazzo/cpstl/blob/master/cpp/algoritmins/hash/test/Main.cpp

Reply:
@Mouradouchane do you need to implement the test before putting it inside the master, tell me if you need help